ci: add netlify preview

dshemetov opened the pull request:
### Checklist

Please:

- [x] Make sure this PR is against "dev", not "main" (unless this is a release
      PR).
- [x] Request a review from one of the current main reviewers:
      brookslogan, nmdefries.
- [x] Makes sure to bump the version number in `DESCRIPTION`. Always increment
      the patch version number (the third number), unless you are making a
      release PR from dev to main, in which case increment the minor version
      number (the second number).
- [x] Describe changes made in NEWS.md, making sure breaking changes
      (backwards-incompatible changes to the documented interface) are noted.
      Collect the changes under the next release number (e.g. if you are on
      1.7.2, then write your changes under the 1.8 heading).
- See [DEVELOPMENT.md](DEVELOPMENT.md) for more information on the development
  process.

### Change explanations for reviewer

This adds the docs preview command: /preview-docs to this package. 

### Magic GitHub syntax to mark associated Issue(s) as resolved when this is merged into the default branch

- Resolves #{issue number}

dshemetov commented on .github/workflows/doc-preview.yaml:
```diff
@@ -0,0 +1,55 @@
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          deploy-message: "Deploy from GitHub Actions"
+          enable-github-deployment: false
+          enable-commit-status: false
```
```suggestion
          enable-commit-status: false
          enable-commit-comment: false
```

brookslogan commented on .github/workflows/doc-preview.yaml:
```diff
@@ -0,0 +1,56 @@
+
+    runs-on: ubuntu-latest
+    permissions:
+      pull-requests: write
+      contents: write
```
question: what are these permissions for?  [This example](https://github.com/nwtgck/actions-netlify?tab=readme-ov-file#usage) doesn't seem to need to add permissions at all, and neither do any of our existing workflows somehow?  (Maybe something connected with [this](https://docs.github.com/en/actions/security-for-github-actions/security-guides/automatic-token-authentication#permissions-for-the-github_token)?  But I thought at some point we did need to specify permissions:...)  And I would have expected some middle ground, like `pull-requests: write` to POST a comment and `contents: read` for the checkout.

dshemetov replied:
That example has `enable-pull-request-comment: false` and we don't, so yea you're right we need `pull-requests: write` write the comment and we probably only need `contents: read`, let me test in the other repo.

brookslogan reviewed (approved):
This looks really cool.

Workflow file looks probably good as far as I can tell, but I'm not too familiar with GitHub Actions and don't know Netlify.  I tried to limit my questions down below but you can skip answering if you'd like, or alternatively, put answers/reminders in comments in the workflow file.

One question not to skip is about whether this could potentially clobber anything valuable on Netlify.  I don't know how to check that so maybe you could double-check somehow?

suggestion: add a mention of how to trigger this in DEVELOPMENT.md or the pull request template?

dshemetov commented on .github/workflows/doc-preview.yaml:
```diff
@@ -0,0 +1,56 @@
+        uses: nwtgck/actions-netlify@v3.0
+        with:
+          publish-dir: './docs'
+          production-deploy: false
```
i have three separate sites for forecasting, epiprocess, and epipredict. all will be separate and unclobbered.

dshemetov commented on .github/workflows/doc-preview.yaml:
```diff
@@ -0,0 +1,56 @@
+
+      - uses: actions/checkout@v4
+        with:
+          ref: refs/pull/${{ steps.pr.outputs.number }}/head
```
You can inline it, just easier for debugging to have the PR_NUM available in the logs.

dshemetov commented:
Thanks for looking into this! I 
- reduced the CI permissions
- removed the extra PR_NUM step
- added lots of comments for all the options

As for the clobbering question: the Netlify sites all have separate SITE_ID which points to different sites on my Netlify account, so there won't be clobbering.
